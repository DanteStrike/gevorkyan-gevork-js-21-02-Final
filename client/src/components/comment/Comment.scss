.comment {
  width: 100%;
  max-height: 250px;
  padding: 5px;
  background-color: var(--primary-background);
  border: var(--primary-border);
  border-radius: 15px;

  &__top {
    display: flex;
    border-bottom: var(--primary-border);
  }

  &__avatar {
    align-self: center;
    margin-right: 5px;
  }

  &__title {
    padding: 0;
    line-height: 1;
    color: var(--primary-font-color);
  }

  &__date {
    font-size: 12px;
    line-height: 1;
    color: var(--secondory-font-color);
  }

  &__text {
    padding: 5px;
    line-height: 1.1;
    color: var(--primary-font-color);
  }
}
