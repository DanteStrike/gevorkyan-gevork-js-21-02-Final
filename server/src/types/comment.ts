import {IUserPreview} from './user';

export interface IComment {
  id: string;
  message: string;
  owner: IUserPreview;
  post: string;
  publishDate: string;
}
