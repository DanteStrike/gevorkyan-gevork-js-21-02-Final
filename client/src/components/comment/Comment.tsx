import React from 'react';
import {IComment} from '../../types';
import LinkedAvatar from '../linked-avatar/LinkedAvatar';
import CustomLink from '../custom-link/CustomLink';
import './Comment.scss';
import {DataUtils, RouteUtils} from '../../utils';

interface ICommentProps {
  comment: IComment;
}

function Comment({comment}: ICommentProps) {
  const {owner, publishDate, message} = comment;
  const {firstName, lastName, title} = owner;
  const name = DataUtils.collectFullName(firstName, lastName, title);
  const profileRoute = RouteUtils.createProfileRoute(owner.id);

  return (
    <article className="comment">
      <div className="comment__top">
        <LinkedAvatar user={owner} className="comment__avatar" to={profileRoute} />
        <div className="comment__wrap">
          <CustomLink to={profileRoute}>
            <h2 className="comment__title">{name}</h2>
          </CustomLink>
          <p className="comment__date">{publishDate}</p>
        </div>
      </div>
      <p className="comment__text">{message}</p>
    </article>
  );
}

export default React.memo(Comment);
