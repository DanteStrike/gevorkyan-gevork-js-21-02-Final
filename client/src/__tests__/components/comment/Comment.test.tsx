import {shallow} from 'enzyme';
import React from 'react';
import {commentOne} from '../../mocks/comment.mock';
import Comment from '../../../components/comment/Comment';

describe(`Component Comment should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Comment comment={commentOne} />);
    expect(wrapper).toMatchSnapshot();
  });
});
