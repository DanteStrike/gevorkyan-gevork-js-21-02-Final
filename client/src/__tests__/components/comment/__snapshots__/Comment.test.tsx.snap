// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Comment should render correctly snapshot default test 1`] = `
<article
  className="comment"
>
  <div
    className="comment__top"
  >
    <Memo(LinkedAvatar)
      className="comment__avatar"
      to="/profile/userID_1"
      user={
        Object {
          "firstName": "anyFN",
          "id": "userID_1",
          "lastName": "antLN",
          "picture": "url",
          "title": "ms",
        }
      }
    />
    <div
      className="comment__wrap"
    >
      <Memo(CustomLink)
        to="/profile/userID_1"
      >
        <h2
          className="comment__title"
        >
          Ms. antLN anyFN
        </h2>
      </Memo(CustomLink)>
      <p
        className="comment__date"
      >
        un-normalized
      </p>
    </div>
  </div>
  <p
    className="comment__text"
  >
    message
  </p>
</article>
`;
